devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -2830,6 +2830,101 @@ bool WholeBodyDynamicsDevice::setupCalibrationWithExternalWrenchesOnTwoFrames(co
+
+    // Clear the class
+    calibrationBuffers.assumedContactLocationsForCalibration.clear();
+
+    // Check if the iRonCub-Mk1 jets frames exist
+    std::string leftArmJetFrame = {"l_arm_jet_turbine"};
+    std::string RightArmJetFrame = {"r_arm_jet_turbine"};
+    std::string leftBackJetFrame = {"chest_l_jet_turbine"};
+    std::string RightBackJetFrame = {"chest_r_jet_turbine"};
+    iDynTree::FrameIndex frameLAIndex = estimator.model().getFrameIndex(leftArmJetFrame);
+    if( frameLAIndex == iDynTree::FRAME_INVALID_INDEX )
+    {
+        yError() << "wholeBodyDynamics : setupCalibrationWithVerticalForcesOnTheFeetAndJetsONiRonCubMk1 impossible to find frame " << leftArmJetFrame << ", Are you using iRonCub-Mk1?";
+        return false;
+    }
+
+    iDynTree::FrameIndex frameRAIndex = estimator.model().getFrameIndex(RightArmJetFrame);
+    if( frameRAIndex == iDynTree::FRAME_INVALID_INDEX )
+    {
+        yError() << "wholeBodyDynamics : setupCalibrationWithVerticalForcesOnTheFeetAndJetsONiRonCubMk1 impossible to find frame " << RightArmJetFrame << ", Are you using iRonCub-Mk1?";
+        return false;
+    }
+
+    iDynTree::FrameIndex frameLBIndex = estimator.model().getFrameIndex(leftBackJetFrame);
+    if( frameLBIndex == iDynTree::FRAME_INVALID_INDEX )
+    {
+        yError() << "wholeBodyDynamics : setupCalibrationWithVerticalForcesOnTheFeetAndJetsONiRonCubMk1 impossible to find frame " << leftBackJetFrame << ", Are you using iRonCub-Mk1?";
+        return false;
+    }
+
+    iDynTree::FrameIndex frameRBIndex = estimator.model().getFrameIndex(RightBackJetFrame);
+    if( frameRBIndex == iDynTree::FRAME_INVALID_INDEX )
+    {
+        yError() << "wholeBodyDynamics : setupCalibrationWithVerticalForcesOnTheFeetAndJetsONiRonCubMk1 impossible to find frame " << RightBackJetFrame << ", Are you using iRonCub-Mk1?";
+        return false;
+    }
+
+    // Check if the soles frames exist
+    std::string LeftSoleFrame = {"l_sole"};
+    std::string RightSoleFrame = {"r_sole"};
+    iDynTree::FrameIndex frameLSIndex = estimator.model().getFrameIndex(LeftSoleFrame);
+    if( frameLSIndex == iDynTree::FRAME_INVALID_INDEX )
+    {
+        yError() << "wholeBodyDynamics : setupCalibrationWithVerticalForcesOnTheFeetAndJetsONiRonCubMk1 impossible to find frame " << LeftSoleFrame;
+        return false;
+    }
+
+    iDynTree::FrameIndex frameRSIndex = estimator.model().getFrameIndex(RightSoleFrame);
+    if( frameRSIndex == iDynTree::FRAME_INVALID_INDEX )
+    {
+        yError() << "wholeBodyDynamics : setupCalibrationWithVerticalForcesOnTheFeetAndJetsONiRonCubMk1 impossible to find frame " << RightSoleFrame;
+        return false;
+    }
+
+    // We assume that all the 6 contacts are Pure Forces with Known Directions (1D) at the origin of each frame
+    iDynTree::Direction zPosAxis = iDynTree::Direction(0,0,1);
+    iDynTree::Direction zNegAxis = iDynTree::Direction(0,0,-1);
+    double p100ThrustN = -2;
+    double p220ThrustN = -9;
```

Comment:
can we just add a comment saying why these values are assigned?

Reply:
Added the comments in https://github.com/robotology/whole-body-estimators/pull/113/commits/9d61f7d9e02c05e86ae517e98ad0a762b1f9f6dc